### ISSUE TITLE
src/log.c: resource leak found by coverity

### ISSUE THREAD

**Issue by chipitsine** [haproxy/haproxy#392]
```
687                        case LF_TEXT:
688                        case LF_SEPARATOR:
   43. noescape: Resource sp is not freed or pointed-to in add_to_logformat_list. [show details]
   44. noescape: Resource str is not freed or pointed-to in add_to_logformat_list. [show details]
   45. Condition !add_to_logformat_list(sp, str, pformat, list_format, err), taking true branch.
689                                if (!add_to_logformat_list(sp, str, pformat, list_format, err))
   46. leaked_storage: Variable var going out of scope leaks the storage it points to.
   47. leaked_storage: Variable backfmt going out of scope leaks the storage it points to.
   48. leaked_storage: Variable str going out of scope leaks the storage it points to.
   CID 1399773 (#2-15 of 15): Resource leak (RESOURCE_LEAK)49. leaked_storage: Variable sp going out of scope leaks the storage it points to.
690                                        return 0;
691                                break;
692                        }
```

**Comment by wtarreau** [comment]
Interesting. In fact it's not particularly this line but all lines with
a "return 0" which leak memory (reason why I hate seeing return in the
middle of a function). These are error cases so the program will exit
anyway. But the function could be cleaned to have a return 0 statement
at the end or a "ret" value changed by error handling to make sure we
always pass through the final free() call.

**Comment by chipitsine** [comment]
I agree it is not very important to free on exit. However, it make some mess in code analysis. We need to keep and eye on "what is important" and "what is not".

Coverity still sees tens of leak (many of them on exit path, but few might be not)

**Comment by chipitsine** [comment]
22 "leaks" are detected by Coverity

**Comment by wtarreau** [comment]
Ilya, I totally agree with you on keeping the code clean on error paths.
Without this we'll always remain limited in what we can call at runtime.
I was just meaning it's not critical. But I'd rather see it fixed anyway.

**Comment by wtarreau** [comment]
OK fixed as far as 1.7 (first one affected), closing now. Thanks Ilya!
